addons/sale/models/payment_transaction.py

```diff
@@ -106,9 +106,9 @@ def _log_message_on_linked_documents(self, message):
         :return: None
         """
         super()._log_message_on_linked_documents(message)
-        self = self.with_user(SUPERUSER_ID)  # Log messages as 'OdooBot'
         for order in self.sale_order_ids or self.source_transaction_id.sale_order_ids:
-            order.message_post(body=message)
+            author = self.env.uid == SUPERUSER_ID and self.env.user.partner_id or order.partner_id
```

Comment:
```suggestion
            author = self.env.user.partner_id if self.env.uid == SUPERUSER_ID else self.partner_id
```
Instead of using the order partner, let's use the tx partner instead as we want to know who effectively made the payment.

Reply:
Hello, @Feyensv Applied suggested changes. Thank you